src/NServiceBus.Core/Receiving/ReceiveCompleted.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Event raised when the transport has completed processing a message.
+    /// </summary>
+    public class ReceiveCompleted
+    {
+        /// <summary>
+        /// Constructs the event.
+        /// </summary>
+        public ReceiveCompleted(string messageId, bool wasAcknowledged, Dictionary<string, string> headers, DateTimeOffset startedAt, DateTimeOffset completedAt)
+        {
+            NativeMessageId = messageId;
+            WasAcknowledged = wasAcknowledged;
+            Headers = headers;
+            StartedAt = startedAt;
+            CompletedAt = completedAt;
+        }
+
+        /// <summary>
+        /// The native message ID.
+        /// </summary>
+        public string NativeMessageId { get; }
+
+        /// <summary>
+        /// True if the message was acknowledged and removed from the queue.
+        /// </summary>
+        public bool WasAcknowledged { get; }
```

Comment:
Yes metrics (and our tests) need to know if the message was acked or not. In the metics case the "MessageProcessed" metrics should not be emitted if the message was rolled back.

>Having said that, we could introduce all that later, and just deprecate this property. Or it could remain as a convenience property which returns true when the outcome was "message was processed" or "message was moved to error queue".

Agree, this can be added later